### PR TITLE
feat: replaced mathematical kwargs in scheduler constructors

### LINKED ISSUE #41: Add `start`/`decay` as Kwargs in Schedulers (apart from the UTF-8 ones)
### Describe the potential feature

When writing code from terminal-based editors e.g. Vim, NeoVim, Nano (and AFAIK Helix), it becomes a bit of a cumbersome to repeatedly copy-paste mathematical characters such as `λ` and `γ` which appear in most schedulers. Although I see the merit of having UTF-8 characters as kwargs for mathematical aesthetics, I believe we should also have the option to specify such parameters by spelling out their meanings i.e. with `start`, `decay`, etc.

As an example, currently we have the following:

```
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.3 (2022-05-06)
 _/ |\__'_|_|_|\__'_|  |  Fedora 36 build
|__/                   |

julia> using ParameterSchedulers

julia> s = Exp(; start = 0.01, decay = 0.95)
ERROR: UndefKeywordError: keyword argument λ not assigned
Stacktrace:
 [1] top-level scope
   @ REPL[2]:1
 [2] top-level scope
   @ ~/.julia/packages/CUDA/DfvRa/src/initialization.jl:52
```

### Motivation

_No response_

### Possible Implementation

I believe that the implementation should be straight forward, just requiring another definition of the schedulers using the `start` and `decay` kwargs (which are already in the "Arguments" section of some of the schedulers doc pages). Using the `Step` scheduler as an example, we'd have:

```julia
Step(;λ, γ, step_sizes) = Step(λ, γ, step_sizes) # current implementation
Step(;start, decay, step_sizes) = Step(start, decay, step_sizes)
```

### PR DESCRIPTION
**Pull request by lfenzo**
Closes #41.

For cyclic schedulers, I've replaced `λ*` with `l*`, but I'm open to suggestions. At this moment I preferred replacing only the symbols instead of making more drastic changes to the API. Tests and docs were also adapted!

cc.: @darsnack

### CODE REVIEW

**Comment by lfenzo** [comment]
Good to go?

File: docs/make.jl [hunk 1]
```diff
@@ -65,10 +65,15 @@ makedocs(;
...
+    )
 )
 
-deploydocs(; repo = "github.com/FluxML/ParameterSchedulers.jl.git", target = "build",
-             push_preview = true)
+deploydocs(;
+    repo = "github.com/FluxML/ParameterSchedulers.jl.git",
+    target = "build",
+    push_preview = true
+)
```
Comment (darsnack) [inline docs/make.jl]: It looks like your formatter is making changes that aren't necessary. Can you revert these?

**Review by darsnack — CHANGES_REQUESTED** [review]
I'm happy with your chosen naming scheme, and most of the changes here are broadly good to go.

I would prefer to release a patch with deprecations first. Can we do something like this:
```
function depkwargs(fn::Symbol, kwargs, remaps::Pair...)
    remaps = Dict(remaps...)
    kwargs = map(keys(kwargs)) do kw
        if haskey(remaps, kw)
            Base.depwarn("Keyword $kw is deprecated. Replacing with $(remaps[kw]) instead.", fn)
            return remaps[kw] => kwargs[kw]
        else
            return kw => kwargs[kw]
        end
    end

    return (; kwargs...)
end

function Triangle(; kwargs...)
     kwargs = depkwargs(:Triangle, kwargs, :λ0 => :l0, :λ1 => :l1)
     return Triangle(kwargs.l0, kwargs.l1, kwargs.period)
end
```

**Comment by lfenzo** [comment]
> `function depkwargs(fn::Symbol, kwargs, remaps::Pair...)` (..) `end`

Very clever! Added to all constructors modified in this PR. Thanks for the snippet!

File: docs/make.jl (hunk 1 above)
Comment (lfenzo) [reply docs/make.jl]: The "formatter" in this case is the human in front of the keyboard :laughing:. I have no problem reverting these, however I must say I prefer the new version, but that mostly concerns aesthetics.

Comment (darsnack) [reply docs/make.jl]: I would revert them cause it just messes with the git blame

File: src/cyclic.jl [hunk 2]
```diff
@@ -1,102 +1,111 @@
...
+    kwargs = depkwargs(:TriangleDecay2, kwargs, :λ0 => :l0, :λ1 => :l1)
+    l0, l1 = kwargs.l0, kwargs.l1
+    return _tridecay2(abs(l0 - l1), min(l0, l1), kwargs.period)
+end
 
 function _tridecay2(range::T, offset, period) where T
     parameters = (Interpolator(Exp(range, T(1/2)), period), offset, period)
-
     return ComposedSchedule(Triangle(range, offset, period), parameters)
 end
```
Comment (darsnack) [inline src/cyclic.jl]: I realized that now the body of `_tridecay2` can be the positional constructor, and the keyword constructor forwards to the positional constructor. Do you mind making this change here and elsewhere?

**Review by darsnack — CHANGES_REQUESTED** [review]
One more small change and then we are good to go. Can you also bump the patch version?

File: docs/make.jl (hunk 1 above)
Comment (lfenzo) [reply docs/make.jl]: Sure!

File: src/cyclic.jl (hunk 2 above)
Comment (lfenzo) [reply src/cyclic.jl]: Done!

**Review by darsnack — APPROVED** [review]
Looks good, thank you! The nightly failure is unrelated to the PR.

**Comment by lfenzo** [comment]
I'm still trying to figure out what happened here, as it turns out, my force push after a `git reset` to revert the formatting in `docs/make.jl` somehow caused this PR to be closed. 

The body of `_tridecay2` and its variants were transferred to the respective positional constructors; patch number was updated and CI errors with Julia nightly seem unrelated.

**Pull request merged**